aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -545,6 +545,235 @@ Tensor& orgqr_helper_cuda_lib(Tensor& result, const Tensor& tau, Tensor& infos,
+  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+
+  cublasFillMode_t uplo = upper ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER;
+  cusolverEigMode_t jobz = compute_eigenvectors ? CUSOLVER_EIG_MODE_VECTOR : CUSOLVER_EIG_MODE_NOVECTOR;
+
+  int n = cuda_int_cast(vectors.size(-1), "n");
+  int lda = std::max<int>(1, n);
+  int batch_size = cuda_int_cast(batchCount(vectors), "batch_size");
+
+  auto vectors_data = vectors.data_ptr<scalar_t>();
+  auto values_data = values.data_ptr<value_t>();
+  auto infos_data = infos.data_ptr<int>();
+
+  // syevj_params controls the numerical accuracy of syevj
+  // by default the tolerance is set to machine accuracy
+  // the maximum number of iteration of Jacobi method by default is 100
+  // cuSOLVER documentations says: "15 sweeps are good enough to converge to machine accuracy"
+  // LAPACK has SVD routine based on similar Jacobi algorithm (gesvj) and there a maximum of 30 iterations is set
+  // Let's use the default values for now
+  syevjInfo_t syevj_params;
+  TORCH_CUSOLVER_CHECK(cusolverDnCreateSyevjInfo(&syevj_params));
+  TORCH_CUSOLVER_CHECK(cusolverDnXsyevjSetSortEig(syevj_params, 1));
+
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+
+  // get the optimal work size and allocate workspace tensor
+  int lwork;
+  at::cuda::solver::syevjBatched_bufferSize<scalar_t>(
+      handle,
+      jobz,
+      uplo,
+      n,
+      vectors_data,
+      lda,
+      values_data,
+      &lwork,
+      syevj_params,
+      batch_size);
+
+  // allocate workspace storage on device
+  auto& allocator = *at::cuda::getCUDADeviceAllocator();
+  auto work_data = allocator.allocate(sizeof(scalar_t) * lwork);
+  at::cuda::solver::syevjBatched<scalar_t>(
+      handle,
+      jobz,
+      uplo,
+      n,
+      vectors_data,
+      lda,
+      values_data,
+      static_cast<scalar_t*>(work_data.get()),
+      lwork,
+      infos_data,
+      syevj_params,
+      batch_size);
+  TORCH_CUSOLVER_CHECK(cusolverDnDestroySyevjInfo(syevj_params));
+}
+
+static void linalg_eigh_cusolver_syevd(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos, bool upper, bool compute_eigenvectors) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(eigenvectors.scalar_type(), "linalg_eigh_cuda", [&] {
+    apply_syevd<scalar_t>(eigenvalues, eigenvectors, infos, upper, compute_eigenvectors);
+  });
+}
+
+static void linalg_eigh_cusolver_syevj(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos, bool upper, bool compute_eigenvectors) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(eigenvectors.scalar_type(), "linalg_eigh_cuda", [&] {
+    apply_syevj<scalar_t>(eigenvalues, eigenvectors, infos, upper, compute_eigenvectors);
+  });
+}
+
+void linalg_eigh_cusolver(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos, bool upper, bool compute_eigenvectors) {
+  // TODO: syevj_batched should be added here, but at least for CUDA 11.2 it contains a bug leading to incorrect results
```

Comment:
@xwang233 are we tracking this, too?

Reply:
Yes, it's the third row in the table here https://github.com/pytorch/pytorch/issues/53879